test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"context"
+	"crypto/sha1" // #nosec G505: not used for security purposes
+	"encoding/hex"
+	"fmt"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+	"antrea.io/antrea/pkg/features"
+	. "antrea.io/antrea/test/e2e/utils"
+)
+
+const (
+	interfaceNameLength = 5
+	nameSpace           = "vm-ns"
+	serviceAccount      = "vm-agent"
+)
+
+type vmInfo struct {
+	nodeName string
+	osType   string
+	ifName   string
+	ip       string
+	ifIndex  string // Used only for Windows
+}
+
+// TestVMAgent is the top-level test which can contain some subtests for
+// VMAgent so they can share setup, teardown.
+func TestVMAgent(t *testing.T) {
+	skipIfFeatureDisabled(t, features.ExternalNode, false, true)
```

Comment:
I feel like this is not sufficient. What if I enable the feature gate (or maybe the feature gate graduates to beta, so it's enabled by default), but I want to run the "normal" tests, and I don't provide a list of VMs? Shouldn't the tests be skipped in this case?

Reply:
This test is specific to VM use-case and it is not testing anything on the cluster. 
Do you have any recommendations on how we could disable/skips the tests if its not required

Reply:
Why not skip if the VM list is empty?

Reply:
Sure. I will add the required change.